src/processes/routing/types.ts

```diff
@@ -30,5 +30,18 @@ export type ServicesListScreenStackProps = NativeStackScreenProps<PaymentsStackP
 export type PaymentScreenStackProps = NativeStackScreenProps<PaymentsStackParamList, 'Payment'>
 export type SuccessOfOperationScreenStackProps = NativeStackScreenProps<PaymentsStackParamList, 'SuccessOfOperation'>
 
+export type AuthStackParamList = {
+    PhoneNumber: undefined,
+    OTPCode: undefined,
+    Password: undefined,
+    ErrorScreen: undefined,
+    SuccessScreen: undefined,
+}
+
+export type PhoneNumberScreenStackProps = NativeStackScreenProps<AuthStackParamList, 'PhoneNumber'>
+export type OTPCodeScreenStackProps = NativeStackScreenProps<AuthStackParamList, 'OTPCode'>
+export type PasswordScreenStackProps = NativeStackScreenProps<AuthStackParamList, 'Password'>
+export type ErrorScreenStackProps = NativeStackScreenProps<AuthStackParamList, 'ErrorScreen'>
+export type SuccessScreenStackProps = NativeStackScreenProps<AuthStackParamList, 'SuccessScreen'>
```

Comment:
Можно убрать дублирование
```suggestion
export type AuthStackParamList = {
    PhoneNumber: undefined,
    OTPCode: undefined,
    Password: undefined,
    ErrorScreen: undefined,
    SuccessScreen: undefined,
}

type TAuthScreenProps<T extends keyof AuthStackParamList> =
  NativeStackScreenProps<TProtectedStackParamsList, T>;

export type PhoneNumberScreenStackProps = TAuthScreenProps<'PhoneNumber'>
export type OTPCodeScreenStackProps = TAuthScreenProps<'OTPCode'>
export type PasswordScreenStackProps = TAuthScreenProps<'Password'>
export type ErrorScreenStackProps = TAuthScreenProps<'ErrorScreen'>
export type SuccessScreenStackProps = TAuthScreenProps<'SuccessScreen'>
```